src/ProjectSystemTools/BuildLogging/Model/Frontend/FrontendBuildTableDataSource.cs

```diff
@@ -37,45 +42,132 @@ internal sealed class FrontEndBuildTableDataSource : ITableEntriesSnapshotFactor
 
         public int CurrentVersionNumber { get; private set; }
 
-        [ImportingConstructor]
-        public FrontEndBuildTableDataSource(IBuildLoggerService loggerService)
+        private static FrontEndBuildTableDataSource selfReference { get; set; }
```

Comment:
This is a potentially problematic pattern. If you ever did need more than one instance of this class, you'd have a bug whereby only the last one created would update.

This is only used in an event handler. Looking at the code, I don't believe that handler needs to be static (it could be a lambda expression too). I think you can remove this field.